### ISSUE TITLE
Incorrect type error with kebab-cased "aria-*" props

### ISSUE THREAD

**Issue by scottbedard** [vuejs/language-tools#4219]
### Vue version

3.4.21

### Link to minimal reproduction

https://play.vuejs.org/#eNqlk19v0zAUxb/KxS8DaUnFn6cqTGJTkUAIJthjJOQ6t6lXx7aunaxo6nfn2mlDB1sf2Jvte/zzyb0n9+KD9+XQo5iLKijSPkLA2Hsw0rbvaxFDLS5qqzvvKMJH52BFroOzcsbrdPFsKl5Kmoq83her2YhlCG8idt7IiLwDqF4UBWxwKZeFkgHBk2t6hQGkBSRyBEUxCtO7krQsjFyiSbZ+eRw1tYBZZu95SnZorhLuztGGWQFw61FFbP7BfTnQ1jqM8r9pLq6Rjjw2YJk/ih9FpyYE1yEjbVugCXjgSxPcg0eq2VE3xDm3Wjm70m15G5zledwnYC2U67w2SN981M7y5TnkSqpJY9zd53wWqcfzw7lao9o8cn4btumsFteEAWnAWky1KKnFOJYXP77iltdTsePJGFafKH7H4EyfPI6yy942bPtIl91+ymHh5tyExTaiDYePSkaTcpf1teD4XJ349D9235bv8r3a7riL+1ieyjO0aJG04u1NTneDK23xmpwPVX5gCsccQiQ2y+yLl69Op7nRAwxF5N4weCI8Mez9D/I8m1PQFpyz/7b6gPKE3Z8DUpoU++V2l29ei91v4ClrVA==

### Steps to reproduce

1. Create a component with a prop name of "ariaLabel"
2. Render that component with a valid value as `aria-label`

### What is expected?

No type errors should be thrown

### What is actually happening?

The kebab-cased version of an `aria-*` prop will produce a type error. This is true of all aria properties, not just valid ones. For example, `aria-whoops` will also produce an error.

```
Argument of type '{ "aria-label": string; }' is not assignable to parameter of type '{ ariaLabel: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Record<string, unknown>'.
  Property 'ariaLabel' is missing in type '{ "aria-label": string; }' but required in type '{ ariaLabel: string; }'.
```

### System Info

```shell
System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 740.64 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
    pnpm: 8.7.1 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 123.0.6312.87
    Safari: 16.3
  npmPackages:
    vue: 3.4.21 => 3.4.21
```

### Any additional comments?

A current workaround is to use a camelCase version of the prop name.

**Comment by johnsoncodehk** [comment]
This is expected, you can set an empty array for `htmlAttributes` in tsconfig to change the behavior.

```
{
  "compilerOptions": {
    // ...
  },
  "vueCompilerOptions": {
    "htmlAttributes": []
  },
}
```

**Comment by scottbedard** [comment]
Gotcha, thanks for the info!

**Comment by CernyMatej** [comment]
This doesn't seem to work for me in Nuxt. Am I doing something wrong?
<img width="283" alt="image" src="https://github.com/vuejs/language-tools/assets/112722215/31ca0780-5380-4602-a0c9-53b3a42eab4c">

**Comment by so1ve** [comment]
You should modify `tsconfig.json` directly, not nuxt config.

**Comment by scottbedard** [comment]
Can anyone explain what the pro/cons are to changing `htmlAttributes`? The tsconfig documentation reads like this,

> **htmlAttributes:** A glob matcher array that should always be recognizing as HTML Attributes rather than Component props. Attribute name will never convert to camelize case.

What does an empty array mean in this context?

**Comment by ihardyslide** [comment]
An explanation would be very nice, since I do not want to add this to my production config without knowing what it really does.
Indeed it fixes the issue, but I do not understand why.

**Comment by ShGKme** [comment]
> What does an empty array mean in this context?

> An explanation would be very nice, since I do not want to add this to my production config without knowing what it really does.
> Indeed it fixes the issue, but I do not understand why.

Initially it was added to allow `kebab-case` for HTML attributes typed with `camelCase`

https://github.com/vuejs/language-tools/issues/1871

The default value for this property is `aria-*`. So only `aria-*` attrs like `aria-label` is affected.

https://github.com/vuejs/language-tools/blob/v2.2.10/packages/language-core/lib/utils/ts.ts#L289

And then it is used here https://github.com/vuejs/language-tools/blob/v2.2.10/packages/language-core/lib/codegen/template/elementProps.ts#L376

Theoretically, it is supposed to break on HTML elements.

But it seems to me that after updating types in Vue, this is not actual anymore, so disabling this `camelize` breaks nothing. If I didn't miss something...

https://github.com/vuejs/core/pull/587/
